File: backends/node/src/dbcontext.js
const SQL = 'sql';
const MONGO = 'mongo';

const store = { dbType: SQL };

export function setDbType(value) {
  store.dbType = value && value.toLowerCase() === MONGO ? MONGO : SQL;
}

export function getDbType() {
  return store.dbType;
}

export { SQL, MONGO };

File: backends/python/app/dbcontext.py
from contextvars import ContextVar

_db_type: ContextVar[str] = ContextVar("db_type", default="sql")

SQL = "sql"
MONGO = "mongo"


def set_db_type(value: str | None) -> None:
    if value and value.lower() == MONGO:
        _db_type.set(MONGO)
    else:
        _db_type.set(SQL)


def get_db_type() -> str:
    return _db_type.get()
